crates/ark/src/sys/windows/interface.rs

```diff
@@ -67,6 +70,29 @@ pub fn setup_r(mut _args: Vec<*mut c_char>) {
         // R_DefParamsEx(params, bindings::RSTART_VERSION as i32);
         R_DefParamsEx(params, 0);
 
+        // Set up "common" command line arguments, inheriting R's "last flag
+        // wins" behavior for these. On the Unix side this is automatically
+        // called by `Rf_initialize_R()`. On the Windows side this is called by
+        // `cmdlineoptions()`, but because we call `R_SetParams()` later on to
+        // tweak some options, we have to fully rebuild the correct `params`
+        // list anyways so we don't supply any user `args` to
+        // `cmdlineoptions()` and instead pass them here.
+        //
+        // Notably sets:
+        // - `(*params).R_Quiet` via `--silent`, `--quiet`, `-q`, `--no-echo`
+        // - `(*params).R_Verbose` via `--verbose`
+        // - `(*params).NoRenviron` via `--no-environ`, `--vanilla`
+        // - `(*params).SaveAction` via `--save`, `--no-save`, `--vanilla`, `--no-echo`
+        // - `(*params).RestoreAction` via `--restore`, `--no-restore`, `--no-restore-data`, `--vanilla`
+        // - `R_RestoreHistory` (a global) via `--restore`, `--no-restore`, `--no-restore-history`, `--vanilla`
+        let mut c_args = RMain::build_ark_c_args(args);
+        let mut c_args_len = c_args.len() as std::ffi::c_int;
+        R_common_command_line(
+            &mut c_args_len,
+            c_args.as_mut_ptr() as *mut *mut c_char,
+            params,
+        );
```

Comment:
We were not respecting any of these on the Windows side. We now respect all of them in a way that should exactly mimic what happens on the Unix side, because we call the same `R_common_command_line()` utility on both sides.